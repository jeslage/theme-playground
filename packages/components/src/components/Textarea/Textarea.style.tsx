import styled from 'styled-components';

const StyledTextarea = styled.label`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: 100%;

  textarea {
    width: 100%;
    height: 80px;
    appeareance: none;
    resize: none;
    background: none;
    padding: 5px;
    max-width: 200px;
    border-radius: 5px;
    border: 1px solid ${({ theme }) => theme.text};
    color: ${({ theme }) => theme.text};
  }
`;

export default StyledTextarea;
