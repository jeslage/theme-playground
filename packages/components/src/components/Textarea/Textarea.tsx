import * as React from 'react';

import StyledTextarea from './Textarea.style';
import Label from '../Label/Label';

export interface Props {
  label: string;
  iconBefore?: JSX.Element;
  description?: string;
  onChange: (val: string) => void;
  value: string;
}

const Textarea: React.FC<Props> = ({
  label,
  iconBefore,
  description,
  value,
  onChange
}) => {
  return (
    <StyledTextarea>
      <Label iconBefore={iconBefore} label={label} description={description} />

      <textarea
        rows={10}
        value={value}
        onChange={event => onChange(event.target.value)}
      />
    </StyledTextarea>
  );
};

export default Textarea;
