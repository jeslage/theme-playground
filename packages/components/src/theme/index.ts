export const darkTheme = {
  base: 'dark',
  backgroundDark: '#2f2f2f',
  backgroundLight: '#333',
  border: '#999999',
  text: '#FFFFFF'
};

export const lightTheme = {
  base: 'light',
  backgroundDark: '#f6f9fc',
  backgroundLight: '#fff',
  border: '#666666',
  text: '#333333'
};

export const getUiTheme = theme => {
  if (theme === undefined) return darkTheme;

  if (typeof theme === 'string') {
    if (theme === 'dark') {
      return darkTheme;
    } else if (theme === 'light') {
      return lightTheme;
    }

    throw Error(
      `Theme "${theme}" is not available. Please use "light" or "dark", or provide a custom theme object.`
    );
  }

  if (theme.base && theme.base === 'light') {
    return {
      ...lightTheme,
      ...theme
    };
  }

  return {
    ...darkTheme,
    ...theme
  };
};
