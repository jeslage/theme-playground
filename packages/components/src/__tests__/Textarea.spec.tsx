import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';

import Textarea from '../components/Textarea/Textarea';
import StyledTextarea from '../components/Textarea/Textarea.style';
import { darkTheme } from '../theme';

describe('Textarea', () => {
  it('Textarea should render correctly', () => {
    const component = shallow(
      <Textarea
        onChange={value => console.log(value)}
        label="Label"
        description="Testbeschreibung"
        value="TestTextarea"
        iconBefore={
          <svg height="24" viewBox="0 0 24 24" width="24">
            <rect height="2" rx="1" width="12" x="6" y="11" />
          </svg>
        }
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('StyledTextarea should render correctly', () => {
    const component = shallow(<StyledTextarea theme={{ colors: darkTheme }} />);

    expect(component).toMatchSnapshot();
  });
});
