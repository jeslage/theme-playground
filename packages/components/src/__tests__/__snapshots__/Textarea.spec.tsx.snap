// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea StyledTextarea should render correctly 1`] = `
<label
  className="sc-AxjAm bkYsvM"
/>
`;

exports[`Textarea Textarea should render correctly 1`] = `
<styled.label>
  <Label
    description="Testbeschreibung"
    iconBefore={
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <rect
          height="2"
          rx="1"
          width="12"
          x="6"
          y="11"
        />
      </svg>
    }
    label="Label"
  />
  <textarea
    onChange={[Function]}
    rows={10}
    value="TestTextarea"
  />
</styled.label>
`;
